Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionCiscoASA.yml

```diff
@@ -0,0 +1,369 @@
+                                                        "Allow");
+      let remainingLogs = allLogs
+                | where DeviceEventClassID in ("106002", "106012", "106013", "106020");
+      let networkaddressWatchlistData = materialize (_ASIM_GetWatchlistRaw("NetworkAddresses"));
+      let internalInterface = networkaddressWatchlistData | where WatchlistItem.Tags has "Internal" | distinct tostring(WatchlistItem["Range Name"]);
+      let externalInterface = networkaddressWatchlistData | where WatchlistItem.Tags has "External" | distinct tostring(WatchlistItem["Range Name"]);
+      union isfuzzy=true all_106001_alike, all_106010_alike, all_106018, all_106023, all_106023_unparsed, all_106023_41,  all_302013_302015_unparsed, all_302013_302015_parsed, all_302014_302016, all_302020_302021, all_7_series, all_106007, all_106017, all_106100_parsed, all_106100_unparsed, remainingLogs
+                | extend 
+                        EventStartTime = TimeGenerated,
+                        EventEndTime = TimeGenerated,
+                        EventVendor = "Cisco",
+                        EventProduct = "ASA",
+                        EventCount = coalesce(EventCount,toint(1)),
+                        EventType = "NetworkSession",
+                        EventSchema = "NetworkSession",
+                        EventSchemaVersion = "0.2.4",
+                        DvcAction = case(trim(@"\s",SimplifiedDeviceAction) in ("Teardown","Built"), "",
+                                        SimplifiedDeviceAction),
+                        SrcInterfaceName = tolower(SrcInterfaceName),
+                        DstInterfaceName = tolower(SrcInterfaceName)
+                | lookup ProtocolLookup on Protocol
+                | project-rename 
+                        EventProductVersion = DeviceVersion,
+                        EventOriginalType = DeviceEventClassID,
+                        EventOriginalSeverity = OriginalLogSeverity,
+                        DvcOriginalAction = DeviceAction,
+                        EventMessage = Message,
+                        Dvc = Computer
+                | extend
+                        EventSeverity = iff(isempty(DvcAction) or DvcAction == "Allow", "Informational", "Low"),                             
+                        DstPortNumber = coalesce(DstPortNumber, DestinationPort),
+                        SrcPortNumber = coalesce(SrcPortNumber, SourcePort),
+                        DstIpAddr = coalesce(DstIpAddr, DestinationIP),
+                        SrcIpAddr = coalesce(SrcIpAddr, SourceIP),
+                        NetworkRuleName = coalesce(NetworkRuleName, DeviceCustomString2),
+                        NetworkDirection = case(isnotempty(CommunicationDirection), CommunicationDirection,
+                                    (SrcInterfaceName in (internalInterface) and DstInterfaceName in (externalInterface)) or DstInterfaceName in (externalInterface), "Outbound",
```

Comment:
The 4th option, both are external interfaces is also valid.

Reply:
@oshezaf What should be the Network Direction in that case?

Reply:
Well, it is in the [doc](https://docs.microsoft.com/en-us/azure/sentinel/network-normalization-schema): External

Reply:
Updated